api/src/test/java/org/pmiops/workbench/api/UserControllerTest.java

```diff
@@ -252,91 +262,212 @@ public void testUserSort() {
     assertThat(robinsonsAsc.getUsers()).containsAllIn(newAscending).inOrder();
   }
 
+  // Combinatorial tests for listBillingAccounts:
+  // enableBillingUpgrade feature flag on/off
+  // free tier available vs. expired
+  // cloud accounts available vs. none
+
+  final BillingAccount freeTierBillingAccount =
+      new BillingAccount()
+          .isFreeTier(true)
+          .displayName("Use All of Us free credits")
+          .name("billingAccounts/free-tier")
+          .isOpen(true);
+
+  final List<com.google.api.services.cloudbilling.model.BillingAccount> cloudbillingAccounts =
+      Lists.newArrayList(
+          new com.google.api.services.cloudbilling.model.BillingAccount()
+              .setName("googlebucks")
+              .setDisplayName("Paid using your credit card"),
+          new com.google.api.services.cloudbilling.model.BillingAccount()
+              .setName("a2")
+              .setDisplayName("Account 2 - Open")
+              .setOpen(true));
+
+  final List<BillingAccount> cloudbillingAccountsInWorkbench =
+      Lists.newArrayList(
+          new BillingAccount()
+              .name("googlebucks")
+              .displayName("Paid using your credit card")
+              .isFreeTier(false)
+              .isOpen(false),
+          new BillingAccount()
+              .name("a2")
+              .displayName("Account 2 - Open")
+              .isFreeTier(false)
+              .isOpen(true));
+
+  // billing upgrade is true, free tier is available, cloud accounts exist
+
   @Test
-  public void listBillingAccounts() throws IOException {
+  public void listBillingAccounts_upgradeYES_freeYES_cloudYES() throws IOException {
     config.billing = new BillingConfig();
     config.billing.accountId = "free-tier";
     config.featureFlags.enableBillingUpgrade = true;
 
-    final com.google.api.services.cloudbilling.model.BillingAccount cloudbillingAccount1 =
-        new com.google.api.services.cloudbilling.model.BillingAccount()
-            .setName("googlebucks")
-            .setDisplayName("Paid using your credit card");
+    when(testFreeTierBillingService.userHasRemainingFreeTierCredits(any())).thenReturn(true);
 
-    final com.google.api.services.cloudbilling.model.BillingAccount cloudbillingAccount2 =
-        new com.google.api.services.cloudbilling.model.BillingAccount()
-            .setName("a2")
-            .setDisplayName("Account 2 - Open")
-            .setOpen(true);
+    when(testCloudbilling.billingAccounts().list().execute())
+        .thenReturn(new ListBillingAccountsResponse().setBillingAccounts(cloudbillingAccounts));
 
-    final List<com.google.api.services.cloudbilling.model.BillingAccount> cloudbillingAccounts =
-        Lists.newArrayList(cloudbillingAccount1, cloudbillingAccount2);
+    final List<BillingAccount> expectedWorkbenchBillingAccounts =
+        ListUtils.union(
```

Comment:
nit: I'd probably go for `ImmutableList.copyOf(Iterables.concat(...))` since there's precedence: https://github.com/all-of-us/workbench/search?q=iterables.concat&unscoped_q=iterables.concat

or new ARrayList(freeTier).addAll(cloudBillingAccounts)

We don't currently use collections4 anywhere else (were collections1-3 already taken?)